addons/mail/models/mail_mail.py

```diff
@@ -536,6 +536,27 @@ def _split_by_mail_configuration(self):
             for batch_ids in tools.split_every(batch_size, record_ids):
                 yield mail_server_id, alias_domain_id, smtp_from, batch_ids
 
+    def send_after_commit(self):
+        """Queues the email to be sent after the commit of the current cursor.
+
+        Useful to send an email only if a transaction is successful.
+        """
+        # send immediately on same cursor when testing as we don't want
+        # to actually commit during tests
+        if getattr(threading.current_thread(), 'testing', False):
```

Comment:
It's relied on for testing and I couldn't find a way to get the updated values from post-commit hooks for some reason. Probably because they use a different cursor, unlike precommit hooks.